planner/core/testdata/enforce_mpp_suite_out.json

```diff
@@ -974,5 +974,85 @@
+          "    └─HashAgg_28 1.00 mpp[tiflash]  funcs:sum(Column#9)->Column#7",
+          "      └─ExchangeReceiver_32 1.00 mpp[tiflash]  ",
+          "        └─ExchangeSender_31 1.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "          └─HashAgg_28 1.00 mpp[tiflash]  funcs:count(distinct test.t.c)->Column#9",
+          "            └─ExchangeReceiver_30 1.00 mpp[tiflash]  ",
+          "              └─ExchangeSender_29 1.00 mpp[tiflash]  ExchangeType: HashPartition, Hash Cols: [name: test.t.c, collate: binary]",
+          "                └─HashAgg_26 1.00 mpp[tiflash]  group by:test.t.c, ",
+          "                  └─TableFullScan_13 10000.00 mpp[tiflash] table:t keep order:false, stats:pseudo"
+        ],
+        "Warn": null
+      },
+      {
+        "SQL": "EXPLAIN select count(distinct c), count(a), count(*) from t;",
+        "Plan": [
+          "TableReader_34 1.00 root  data:ExchangeSender_33",
+          "└─ExchangeSender_33 1.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "  └─Projection_27 1.00 mpp[tiflash]  Column#7, Column#8, Column#9",
+          "    └─HashAgg_28 1.00 mpp[tiflash]  funcs:sum(Column#17)->Column#7, funcs:sum(Column#18)->Column#8, funcs:sum(Column#19)->Column#9",
+          "      └─ExchangeReceiver_32 1.00 mpp[tiflash]  ",
+          "        └─ExchangeSender_31 1.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "          └─HashAgg_28 1.00 mpp[tiflash]  funcs:count(distinct test.t.c)->Column#17, funcs:sum(Column#15)->Column#18, funcs:sum(Column#16)->Column#19",
+          "            └─ExchangeReceiver_30 1.00 mpp[tiflash]  ",
+          "              └─ExchangeSender_29 1.00 mpp[tiflash]  ExchangeType: HashPartition, Hash Cols: [name: test.t.c, collate: binary]",
+          "                └─HashAgg_26 1.00 mpp[tiflash]  group by:test.t.c, funcs:count(test.t.a)->Column#15, funcs:count(1)->Column#16",
+          "                  └─TableFullScan_13 10000.00 mpp[tiflash] table:t keep order:false, stats:pseudo"
+        ],
+        "Warn": null
+      },
+      {
+        "SQL": "EXPLAIN select count(distinct c+a), count(a) from t;",
+        "Plan": [
+          "TableReader_34 1.00 root  data:ExchangeSender_33",
+          "└─ExchangeSender_33 1.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "  └─Projection_27 1.00 mpp[tiflash]  Column#7, Column#8",
+          "    └─HashAgg_28 1.00 mpp[tiflash]  funcs:sum(Column#15)->Column#7, funcs:sum(Column#16)->Column#8",
+          "      └─ExchangeReceiver_32 1.00 mpp[tiflash]  ",
+          "        └─ExchangeSender_31 1.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "          └─HashAgg_28 1.00 mpp[tiflash]  funcs:count(distinct Column#13)->Column#15, funcs:sum(Column#14)->Column#16",
+          "            └─ExchangeReceiver_30 1.00 mpp[tiflash]  ",
+          "              └─ExchangeSender_29 1.00 mpp[tiflash]  ExchangeType: HashPartition, Hash Cols: ",
```

Comment:
Hash Cols: here should be filled with `
Column#18`

Reply:
Nice catch, I think it is caused by this issue: https://github.com/pingcap/tidb/issues/35417
I will bail out for cases where distinct column is an expression.

Reply:
ok